pkg/controller/infinispan/infinispan_controller.go

```diff
@@ -157,6 +157,21 @@ func (r *ReconcileInfinispan) Reconcile(request reconcile.Request) (reconcile.Re
 	// Apply defaults if not already set
 	infinispan.ApplyDefaults()
 
+	// Perform all the possible preliminary checks before go on
+	recResult, err := infinispan.PreliminaryChecks()
+	if err != nil {
+		if infinispan.SetCondition("PreliminaryChecksFailed", "true", err.Error()) {
```

Comment:
@rigazilla what do you think about to introduce conditions enum in infinispan type?

Reply:
@dmvolod yeah, I usually tent to formalize things as late as possible, but It's probably time to give a shape to the bunch of string conditions. It would be great to have a separate discussion on this and maybe labels/selectors, to check if we're doing something that can be easily integrated with `oc wait` and maybe other `oc` commands.

Reply:
Yes, agree with you, @rigazilla 
Other things in this PR good for me

Reply:
@rigazilla , looks like this PreliminaryChecksFailed need to start from lowercase or not?